src/Sylius/Component/Promotion/Processor/PromotionProcessor.php

```diff
@@ -45,7 +68,7 @@ public function process(PromotionSubjectInterface $subject)
 
         $eligiblePromotions = [];
 
-        foreach ($this->getActivePromotions($subject) as $promotion) {
+        foreach ($this->activePromotionsProvider->provide() as $promotion) {
```

Comment:
``PreQualifiedPromotionsProvider->getPromotions($subject)`` - It should give us promotions pre qualified for given subject.